Automattic-Tracks-iOS/Event Logging/EventLoggingUploadManager.swift

```diff
@@ -0,0 +1,53 @@
+import Foundation
+import CocoaLumberjack
+import Sodium
+
+class EventLoggingUploadManager {
+
+    private struct Constants {
+        static let uuidHeaderKey = "log-uuid"
+        static let uploadHttpMethod = "POST"
+    }
+
+    var dataSource: EventLoggingDataSource
+    var delegate: EventLoggingDelegate
+    var networkService: EventLoggingNetworkService = EventLoggingNetworkService()
+    var fileManager: FileManager = FileManager.default
```

Comment:
I'd also consider making these `private let`. I find that it's easier to understand how to use a class if it has few public methods. 

```suggestion
    private let dataSource: EventLoggingDataSource
    private let delegate: EventLoggingDelegate
    private let networkService: EventLoggingNetworkService = EventLoggingNetworkService()
    private let fileManager: FileManager = FileManager.default
```

Reply:
Chatting on Slack, we discussed how decreased mutability was preferable to simpler initialization, so these are now all provided via the constructor in a970df2b1c6f215367daf1d784285316a2a90c4d.

Reply:
Perfect 🎉